### DISCUSSION TITLE
Getting Maximum benefit from MacBook pro M2 using AlDente Pro

### DISCUSSION THREAD

**Discussion by Lordsnooty05** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#1351] [General]
I only bought AlDente Pro a couple of days ago and so far so good. I had a 2019 MacBook Pro (Intel processor) and it was plugged into the electricity full time....so it probably was charged to 100% most of the time. I thought that was the way to go and I thought being plugged into the mains electricity in the UK that was fine. Obviously, that was the wrong way and it developed a Battery problem and just couldn't hold a charge once it was unplugged from the mains electricity it had problems rebooting and it advised me to get a replacement battery.
Due to other minor details I decided to buy a new M2 macBook pro and came across your app and tried the free version which was good but I felt I would benefit from the Pro Version which I did a few days ago.
One issue I have had is that on my new MacBook Pro M2 the activation code has de-activated a number of times (about 4 times). It just goes back to the free version and I have to put my code in to reactivate it back to the pro-version. I take it that this issue will not affect the number of times I use AlDente Pro if I ever replace my MacBook Pro in the future
One of the main reasons for purchasing the AlDente pro is to get maximum life out of the battery so can you suggest the best way to run the app to get maximum life....for example, have it plugged into the mains electricity and have it set to 80% and have all or most of the settings on the app set to On. I would appreciate your thoughts on this matter. (I usually put the MacBook Pro into sleep mode at midnight

**Comment by MatthiasKerbl** [comment]
Hi @Lordsnooty05,

Thank you for supporting us in getting AlDente Pro.
Our licensing server provider is currently facing a bug that results in the license being deactivated for some users regularly. This should be fixed soon.

> I take it that this issue will not affect the number of times I use AlDente Pro if I ever replace my MacBook Pro in the future

No, it won't.

Regarding your question about how to get the maximum life out of your battery, this is unfortunately not that easy to answer. First of all, it depends on your personal usage behavior and is, therefore, different for each user. Secondly, there is also no perfect consensus within the research about what the ultimate protocol is. However, there are trends and rules of thumb. In general, using a charge limit of 80% is much better than leaving the battery at 100%, and probably the best charge limit is somewhere between 40% and 60%. Also, keeping the temperature in a healthy range will help a lot. You can use the Heat Protection feature and set it somewhere between 30 and 35 degrees Celsius, depending on your environment, and also try to avoid temperatures below ca. 10 degrees Celsius. Overnight, you can either turn off and unplug the MacBook or leave it plugged in and let it sleep.
Of course, there is still much more to this, but I hope I could help you get started.

Best,
Matthias
